ngsa/src/main/java/com/cse/ngsa/app/services/volumes/VolumeSecretService.java

```diff
@@ -27,6 +31,12 @@ public Secrets getAllSecretsFromVolume(String volume) {
     sec.setCosmosKey(getSecretFromFile(volume,Constants.COSMOS_KEY_KEYNAME));
     sec.setCosmosUrl(getSecretFromFile(volume,Constants.COSMOS_URL_KEYNAME));
 
+    Matcher m = cosmosNamePat.matcher(sec.getCosmosUrl());
+    if (m.matches() && m.groupCount() > 0) {
+      // group 0 --> total match
+      // group 1 --> first group match, we only have one group
```

Comment:
do we need these comments here?

Reply:
No, we do not. Added for clarity and future reference.